src/Process.js

```diff
@@ -1,59 +1,76 @@
-        this.queue = null;
-        this.cpuTimeNeeded = (cpuTimeNeeded !== null) ? cpuTimeNeeded : Math.round(Math.random() * 1000);
-        this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0;
-        // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
-        this.stateChanged = false;
-    }
-    
-    setParentQueue(queue) {
+  constructor(pid, cpuTimeNeeded = null, blocking = false) {
+    this._pid = pid;
+    this.queue = null;
+    this.cpuTimeNeeded =
+      cpuTimeNeeded !== null ? cpuTimeNeeded : Math.round(Math.random() * 1000);
+    this.blockingTimeNeeded = blocking ? Math.round(Math.random() * 100) : 0;
+    // A bool representing whether this process was toggled from blocking to non-blocking or vice versa
+    this.stateChanged = false;
+  }
 
-    }
+  setParentQueue(queue) {
+    this.queue = queue;
+  }
 
-    isFinished() {
+  isFinished() {
+    if (this.cpuTimeNeeded <= 0 && this.blockingTimeNeeded <= 0) return true;
+    return false;
+  }
 
+  // If no blocking time is needed by this process, decrement the amount of
+  // CPU time it needs by the input time
+  // If blocking time is needed by this process, move it to the blocking queue
+  // by emitting the appropriate interrupt
+  // Make sure the `stateChanged` flag is toggled appropriately
+  executeProcess(time) {
+    if (this.blockingTimeNeeded === 0) {
+      this.cpuTimeNeeded -= time;
+      this.cpuTimeNeeded = this.cpuTimeNeeded <= 0 ? 0 : this.cpuTimeNeeded;
+    } else {
+      this.queue.emitInterrupt(this, SchedulerInterrupt.PROCESS_BLOCKED);
+      this.stateChanged = true;
     }
+  }
 
-    // If no blocking time is needed by this process, decrement the amount of 
-    // CPU time it needs by the input time
-    // If blocking time is needed by this process, move it to the blocking queue
-    // by emitting the appropriate interrupt
-    // Make sure the `stateChanged` flag is toggled appropriately
-    executeProcess(time) {
-
-   }
-
-   // If this process requires blocking time, decrement the amount of blocking
-   // time it needs by the input time
-   // Once it no longer needs to perform any blocking execution, move it to the 
-   // top running queue by emitting the appropriate interrupt
-   // Make sure the `stateChanged` flag is toggled appropriately
-    executeBlockingProcess(time) {
+  // If this process requires blocking time, decrement the amount of blocking
+  // time it needs by the input time
+  // Once it no longer needs to perform any blocking execution, move it to the
+  // top running queue by emitting the appropriate interrupt
+  // Make sure the `stateChanged` flag is toggled appropriately
+  executeBlockingProcess(time) {
+    if (this.blockingTimeNeeded > 0) {
+      this.blockingTimeNeeded -= time;
+      this.blockingTimeNeeded =
+        this.blockingTimeNeeded > 0 ? this.blockingTimeNeeded : 0;
 
+      if (this.blockingTimeNeeded === 0) {
```

Comment:
If you choose to check against zero like this, make sure you're never letting a process have additional "negative runtime" after its completion.